### ISSUE TITLE
Keyboard navigation does not work with customInput

### ISSUE THREAD

**Issue by paustint** [Hacker0x01/react-datepicker#1338]
### Expected behavior
1. Calendar with custom input works with keyboard

### Actual behavior
1. Using a custom input, keyboard navigation does not work

### Steps to reproduce
1. usign example page, go to the Custom Input section
1. click on the button to open the calendar
1. press keyboard arrow keys and notice that the browser window scrolls

Code used in example
```javascript
class ExampleCustomInput extends React.Component {

  render () {
    return (
      <button
        className="example-custom-input"
        onClick={this.props.onClick}>
        {this.props.value}
      </button>
    )
  }
}

ExampleCustomInput.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string
};

...

<DatePicker
  customInput={<ExampleCustomInput />}
  selected={this.state.startDate}
  onChange={this.handleChange} />
```

**Comment by ziweiwu** [comment]
this is indeed an issue.

**Comment by VinceSJ** [comment]
Seeing the same thing.

**Comment by sascha-friederich** [comment]
I found a way... the customInput element should has to use the onKeyDown-event (that gets passed to the custom input element by the DatePicker)
```
class ExampleCustomInput extends React.Component {

  render () {
    return (
      <button
       onKeyDown={this.props.onKeyDown}
        className="example-custom-input"
        onClick={this.props.onClick}>
        {this.props.value}
      </button>
    )
  }
}
```

i found this, analyzing the code... i think this should be mentioned in the documentation!
